toolkits/slack/arcade_slack/utils.py

```diff
@@ -174,6 +191,110 @@ def extract_basic_user_info(user_info: SlackUser) -> BasicUserInfo:
     )
 
 
+async def retrieve_conversations_by_user_ids(
```

Comment:
this one is a doozy. Are there any opportunities to flatten this?  👉👈 

Even if it's just defining a _associate_members function outside of this

Reply:
I'll think more tomorrow. Already went through several iterations to get here 😄

Like @Spartee says, gotta stop painting at some point...

Reply:
@EricGustin just pushed a new implementation. it looks less doozy to me now 😄

let me know your thoughts on it